### PR TITLE
d-cryptic/temperature

### PR DESCRIPTION
**Pull request by d-cryptic**
I have changed all the relavent temp variables/keywords to temperature. Though I haven't checked it locally as I am getting some issue. But also all temp are changed to 'temperature'.  Except some temps in interfaces.py

### CODE REVIEW

**Comment by KonradMcClure** [comment]
Hey @d-cryptic! Nice work. There are a few things I think should be added before it's ready to merge:
- Anywhere that says `temperaturecontroller` should probably switch to `temperature_controller` so it's easier to read`. I believe this will also allow it to pass our Spell Checking workflow (see above)
- It looks like import statements were changed without changing the actual file name (i.e. `import temp_probe_mock` to `import temperature_probe_mock`). Go ahead and rename those files as well. This is what caused the Python Tests workflow and Lint Code Base workflow to fail by the looks of it.

Thanks for you contributions!

**Comment by KonradMcClure** [comment]
Once all three of the workflow checks have passed, we should be ready to merge it in.

**Comment by d-cryptic** [comment]
Ok will do the changes oday

**Comment by d-cryptic** [comment]
I have done the mentioned changes, please review it

**Comment by d-cryptic** [comment]
I found the error, let me resolve that

File: src/interfaces.py [hunk 1]
```diff
@@ -15,26 +15,26 @@
...
 import devices.ph_probe_mock as ph_probe_mock
 import devices.serial_mock as serial_mock
-import devices.temp_probe as temp_probe
-import devices.temp_probe_mock as temp_probe_mock
-import devices.tempcontrol as tempcontrol  # Temp
-import devices.tempcontrol_mock as tempcontrol_mock
+import devices.temperature_probe as temperature_probe
+import devices.temperature_probe_mock as temperature_probe_mock
+import devices.temperaturecontrol as temperaturecontrol  # Temperature
+import devices.temperaturecontrol_mock as temperaturecontrol_mock
```
Comment (KonradMcClure) [inline src/interfaces.py:21]: You correctly changed the filenames to `temperature_control` and `temperature_control_mock` however these imports need to match that for them to be able to import

File: src/interfaces.py [hunk 2]
```diff
@@ -15,26 +15,26 @@
...
 
 ph_class: types.ModuleType
-temp_class: types.ModuleType
+temperature_class: types.ModuleType
 board_class: types.ModuleType
 lcd_class: types.ModuleType
 keypad_class: types.ModuleType
-tempcontrol_class: types.ModuleType
+temperaturecontrol_class: types.ModuleType
```
Comment (KonradMcClure) [inline src/interfaces.py:28]: update to temperature_control_class

File: src/interfaces.py [hunk 3]
```diff
@@ -15,26 +15,26 @@
...
 
 if constants.IS_TEST:
     ph_class = ph_probe_mock
-    temp_class = temp_probe_mock
+    temperature_class = temperature_probe_mock
     board_class = board_mock
     lcd_class = lcd_mock
     keypad_class = keypad_mock
-    tempcontrol_class = tempcontrol_mock
+    temperaturecontrol_class = temperaturecontrol_mock
```
Comment (KonradMcClure) [inline src/interfaces.py:37]: update to temperature_control_class

Comment (d-cryptic) [reply src/interfaces.py:37]: I think I made a mistake while pushing the code, I pushed the wrong branch or file. I am working on it

File: src/interfaces.py [hunk 4]
```diff
@@ -43,36 +43,36 @@
     import board  # All hardware (see above note)
 
     ph_class = ph_probe
-    temp_class = temp_probe
+    temperature_class = temperature_probe
     board_class = board
     lcd_class = lcd
     keypad_class = keypad
-    tempcontrol_class = tempcontrol
+    temperaturecontrol_class = temperaturecontrol
```
Comment (KonradMcClure) [inline src/interfaces.py:50]: update to temperature_control_class

File: src/interfaces.py [hunk 5]
```diff
@@ -43,36 +43,36 @@
...
     # LCD and ui_keypad setup
     ui_lcd = setup_lcd()
     ui_keypad = setup_keypad()
 
-    # Temp Control Setup
-    temp_sensor = setup_temp_probe()
-    tempcontroller = setup_tempcontrol()
+    # Temperature Control Setup
+    temperature_sensor = setup_temperature_probe()
+    temperature_controller = setup_temperaturecontrol()
```
Comment (KonradMcClure) [inline src/interfaces.py:75]: setup_temperature_control()

File: src/interfaces.py [hunk 6]
```diff
@@ -109,17 +109,17 @@ def setup_keypad():
...
 
 def setup_temp_probe():
-    return temp_class.Temp_Probe(
+    return temperature_class.Temperature_Probe(
         board_class.SCK, board_class.MOSI, board_class.MISO, board_class.D4, wires=3
     )
 
 
-def setup_tempcontrol():
+def setup_temperaturecontrol():
```
Comment (KonradMcClure) [inline src/interfaces.py]: setup_temperture_control()

**Comment by d-cryptic** [comment]
Done the changes in interfaces.py

**Comment by KonradMcClure** [comment]
Oh sorry I didn't realize it would just spam the conversations here with those. There's a good number of places that still need temperaturecontrol to be changed to temperature_control

**Comment by d-cryptic** [comment]
Hey, I cant find the error. I checked there is no error showing in my commits

**Comment by KonradMcClure** [comment]
Alright I think that looks good! Thank you for doing that! The Spell Check and Lint didn't pass but I think I'll just do a quick update and pull request after merging to take care of those quick.

**Comment by d-cryptic** [comment]
Thanks for all the help, I learned a lot.

**Pull request merged**